### PR TITLE
Free annotations to ingress

### PR DESCRIPTION
**Pull request by Hi-Fi**

### CODE REVIEW

File: kolga/settings.py [hunk 1]
```diff
@@ -90,6 +90,7 @@
     # ================================================
     "K8S_ADDITIONAL_HOSTNAMES": [env.list_none, []],
     "K8S_CLUSTER_ISSUER": [env.str, ""],
+    "K8S_INGRESS_ANNOTATIONS": [env.str, ""],
```
Comment (frwickst) [inline kolga/settings.py]: Here you can use `env.list_none` which will turn the string into a list for you. That would then look like:
`"K8S_INGRESS_ANNOTATIONS": [env.list_none, []],`

That will also take care of the splitting at least for you, it will not do the key/value splitting.

File: kolga/settings.py [hunk 2]
```diff
@@ -172,6 +173,7 @@ class Settings:
     SERVICE_ARTIFACT_FOLDER: str
     K8S_ADDITIONAL_HOSTNAMES: List[str]
     K8S_CLUSTER_ISSUER: str
+    K8S_INGRESS_ANNOTATIONS: str
```
Comment (frwickst) [inline kolga/settings.py]: This would also turn into `K8S_INGRESS_ANNOTATIONS: List[str]`

File: kolga/libs/kubernetes.py [hunk 3]
```diff
@@ -529,6 +529,11 @@ def create_application_deployment(
 
         if settings.K8S_INGRESS_WHITELIST_IPS:
             values["ingress"]["whitelistIP"] = settings.K8S_INGRESS_WHITELIST_IPS
+        
+        if settings.K8S_INGRESS_ANNOTATIONS:
+            for annotation in settings.K8S_INGRESS_ANNOTATIONS.split(","):
```
Comment (frwickst) [inline kolga/libs/kubernetes.py]: The split() call will not be needed if `env.list_none` is used here.

File: docs/pages/variables.md [hunk 4]
```diff
@@ -18,67 +20,64 @@ on other platforms, for that reason we add a variable `GIT_COMMIT_SHA` which alw
 point to the SHA of a commit, no matter the underlying platform, as long as the platform
 supplies a predefined variables with the SHA.
 
-
```
Comment (frwickst) [inline docs/pages/variables.md]: I see that you have removed all escaping from the table here. I guess that is still fine at this point and we don't really need it right this second. However, this does not relate to the commit that you have created and should be split up into a separate commit. (this goes for all documentation formatting changes)

**Review by frwickst — CHANGES_REQUESTED** [review]
Overall this looks like a nice addition, I see no reason not to support this. There are a few things that can be simplified and a bit of git hygiene that would be nice to get fixed here before we merge.

Good work 👍

File: docs/pages/variables.md (hunk 4 above)
Comment (Hi-Fi) [reply docs/pages/variables.md]: IDE actually made the removal, when formatted the table to nice looking version. Have to to the nice layout probably by hand and not by VScode format.

Comment (Hi-Fi) [reply docs/pages/variables.md]: Updated all, let see if PR would realize that also at some point...

File: kolga/settings.py (hunk 1 above)
Comment (Hi-Fi) [reply kolga/settings.py]: Thank you, I saw that but for some reason made self. Probably thought to use some other separator than comma, as separator can't be in the annotation value.
